### DISCUSSION TITLE
about shadowing

### DISCUSSION THREAD

**Discussion by rraammiinn** [vlang/v#17471] [General]
doesn't elimination of shadowing cause any problem ?
if someone use a library doesn't program variables make conflict with variables in that library ?

doesn't it make choosing name for variables harder ?

**Comment by JalonSolov** [comment]
No, it doesn't conflict with modules, as things from modules are prefixed by the module name.

Shadowing leads to all sorts of subtle bugs, where you wind up modifying something different from what you thought you were modifying.  Code is much safer when shadowing isn't allowed.
